Kitodo/src/main/java/org/kitodo/production/security/AESUtil.java

```diff
@@ -0,0 +1,160 @@
+        System.arraycopy(cipherText, 0, cipherCombined, SALT_LENGTH, cipherText.length);
+        System.arraycopy(iv, 0, cipherCombined, SALT_LENGTH + cipherText.length, iv.length);
+
+        return Base64.getEncoder().encodeToString(cipherCombined);
+    }
+
+    /**
+     * Checks if value is encrypted.
+     *
+     * <p>
+     * Function checks if value starts with defined salt prefix after base64
+     * decoding.
+     * </p>
+     *
+     * @param potentialEncryptedValue
+     *            The value to be checked.
+     * @return boolean true if value is encrypted using encrypt function
+     */
+    public static boolean isEncrypted(String potentialEncryptedValue) {
+        try {
+            byte[] cipherCombined = Base64.getDecoder().decode(potentialEncryptedValue);
+            byte[] saltPrefix = Arrays.copyOfRange(cipherCombined, 0, SALT_PREFIX.getBytes().length);
+            // check if cipher combined has salt prefix
+            return Arrays.equals(saltPrefix, SALT_PREFIX.getBytes());
+        } catch (IllegalArgumentException e) {
+            // not valid base64 string so
+        }
+        return false;
+    }
+
+    /**
+     * Decrypts encrypted value by the secret using AES/CBC/PKCS5Padding algorithm.
+     * 
+     * <p>
+     * Function splits encrypted value salt, cipher text and iv after base64
+     * decoding. Salt and secret will be used to build AES-256 secret key. With the
+     * help of the iv and the secret key, the cipher text is decrypted and returned.
+     * </p>
+     *
+     * @param encryptValue
+     *            The encrypted value
+     * @param secret
+     *            The secret from config properties
+     * @return The decrypted value.
+     */
+    public static String decrypt(String encryptValue, String secret)
+            throws NoSuchPaddingException, NoSuchAlgorithmException, InvalidAlgorithmParameterException,
+            InvalidKeyException, BadPaddingException, IllegalBlockSizeException, InvalidKeySpecException {
+        byte[] cipherCombined = Base64.getDecoder().decode(encryptValue);
+
+        // get cipherText and iv from cipherCombined
+        byte[] salt = Arrays.copyOfRange(cipherCombined, 0, SALT_LENGTH);
+        byte[] cipherText = Arrays.copyOfRange(cipherCombined, SALT_LENGTH, cipherCombined.length - IV_LENGTH);
+        byte[] iv = Arrays.copyOfRange(cipherCombined, cipherCombined.length - IV_LENGTH, cipherCombined.length);
+
+        Cipher cipher = Cipher.getInstance(ALGORITHM);
+        cipher.init(Cipher.DECRYPT_MODE, getSecretKey(secret, salt), new IvParameterSpec(iv));
+        byte[] value = cipher.doFinal(cipherText);
+        return new String(value);
+    }
+
+    private static SecretKey getSecretKey(String secret, byte[] salt)
+            throws InvalidKeySpecException, NoSuchAlgorithmException {
+        if (StringUtils.isBlank(secret) || Objects.isNull(salt)) {
+            return null;
```

Comment:
Avoid returning `null` value as `null` checks must be done everywhere this method is used.